### PR TITLE
Bump Calico version

### LINKED ISSUE #3279: Bump Dependencies for MetalK8s 2.10
**Component**:

'build'

**Why this is needed**:

We didn't update dependencies such as Prometheus, Grafana, etc ... since branch 2.8. We want to always embed the latest fix, security fixes, or new features and do it before releasing the first patch version of a specific branch.

**What should be done**:

Check and update all dependencies and make sure we don't break anything ...

**Implementation proposal** (strongly recommended):

**Test plan**:

### PR DESCRIPTION
**Pull request by alexandre-allard**
**Component**: packages, salt, build

**Context**: 

**Summary**:
Bump Calico binaries, images and manifest to v3.18.3

**Acceptance criteria**: 

---

See: #3279

### CODE REVIEW

**Review by NicolasT — DISMISSED** [review]
LGTM, though not a big fan of doing this in 4 commits, since they're not reversible units, individually.

File: salt/metalk8s/kubernetes/cni/calico/deployed.sls [hunk 1]
```diff
@@ -3883,3 +3913,5 @@ spec:
 
 ---
 # Source: calico/templates/configure-canal.yaml
+
+
```
Comment (TeddyAndrieux) [inline salt/metalk8s/kubernetes/cni/calico/deployed.sls:3937]: Oh cool :smile:

**Review by TeddyAndrieux — DISMISSED** [review]

**Comment by alexandre-allard** [comment]
@NicolasT Agree, I just find it easier to review with small commits, I've squashed everything.

File: salt/metalk8s/kubernetes/cni/calico/deployed.sls (hunk 1 above)
Comment (alexandre-allard) [reply salt/metalk8s/kubernetes/cni/calico/deployed.sls:3937]: Haha, I stick to what's upstream.

**Review by TeddyAndrieux — APPROVED** [review]

**Pull request merged**